ledger/src/blockstore.rs

```diff
@@ -3935,20 +3935,18 @@ impl Blockstore {
     /// Returns a mapping from each elements of `slots` to a list of the
     /// element's children slots.
     pub fn get_slots_since(&self, slots: &[Slot]) -> Result<HashMap<Slot, Vec<Slot>>> {
-        let slot_metas: Result<Vec<Option<SlotMeta>>> = self
-            .meta_cf
-            .multi_get(slots.iter().copied())
-            .into_iter()
-            .collect();
-        let slot_metas = slot_metas?;
-
-        let result: HashMap<Slot, Vec<Slot>> = slots
-            .iter()
-            .zip(slot_metas)
-            .filter_map(|(slot, meta)| meta.map(|meta| (*slot, meta.next_slots.to_vec())))
-            .collect();
+        let keys = self.meta_cf.multi_get_keys(slots.iter().copied());
+        let slot_metas = self.meta_cf.multi_get(&keys);
+
+        let mut slots_since: HashMap<Slot, Vec<Slot>> = HashMap::with_capacity(slots.len());
+        for meta in slot_metas.into_iter() {
+            let meta = meta?;
+            if let Some(meta) = meta {
+                slots_since.insert(meta.slot, meta.next_slots);
```

Comment:
is `meta.slot` always the same as what would have been in the corresponding `slots` slice index? I'm wondering why the previous code zipped these two together instead of just iterating slot_metas

Reply:
Yes, the `SlotMeta` stored at slot `X` will have `.meta == X`. The two places where we insert these:
https://github.com/anza-xyz/agave/blob/e48e123ff970ef8896b80eb9d6737d1d06b7ef9d/ledger/src/blockstore.rs#L4656
https://github.com/anza-xyz/agave/blob/e48e123ff970ef8896b80eb9d6737d1d06b7ef9d/ledger/src/blockstore.rs#L4703

> I'm wondering why the previous code zipped these two together instead of just iterating slot_metas

Unsure, maybe the previous author was unaware that `SlotMeta` contains the slot